internal/bminventory/inventory.go

```diff
@@ -365,9 +365,11 @@ func (b *bareMetalInventory) RegisterCluster(ctx context.Context, params install
 	}
 
 	if sshPublicKey := swag.StringValue(&cluster.SSHPublicKey); sshPublicKey != "" {
+		sshPublicKey = strings.TrimSpace(cluster.SSHPublicKey)
```

Comment:
maybe we should move this line outside the if, so we will cover also cases of key=="\n\n"

Reply:
it covers it. I added to if cause if validates that SSHPublicKey is set at all